src/edu/csus/ecs/pc2/ui/NSAStandingsPane.java

```diff
@@ -0,0 +1,423 @@
+            messageLabel.setHorizontalAlignment(javax.swing.SwingConstants.CENTER);
+            messagePane = new JPanel();
+            messagePane.setLayout(new BorderLayout());
+            messagePane.setPreferredSize(new java.awt.Dimension(25, 25));
+            messagePane.add(messageLabel, java.awt.BorderLayout.CENTER);
+        }
+        return messagePane;
+    }
+
+    private void showMessage(final String string) {
+
+        SwingUtilities.invokeLater(new Runnable() {
+            public void run() {
+                messageLabel.setText(string);
+            }
+        });
+    }
+
+    private Object[] buildSiteRow(String[] fields) {
+
+//        Object[] cols = { "Rank", "Name", "Solved", "Points" };
+
+        Object[] obj = new Object[standingsListbox.getColumnCount()];
+
+        obj[0] = fields[0];
+        obj[1] = fields[1];
+        obj[2] = fields[2];
+        obj[3] = fields[3];
+
+        return obj;
+    }
+
+    /**
+     * @author pc2@ecs.csus.edu
+     *
+     */
+    public class AccountListenerImplementation implements IAccountListener {
+
+        public void accountAdded(AccountEvent accountEvent) {
+            refreshStandings();
+        }
+
+        public void accountModified(AccountEvent event) {
+            refreshStandings();
+        }
+
+        public void accountsAdded(AccountEvent accountEvent) {
+            refreshStandings();
+        }
+
+        public void accountsModified(AccountEvent accountEvent) {
+            refreshStandings();
+        }
+
+        public void accountsRefreshAll(AccountEvent accountEvent) {
+            refreshStandings();
+        }
+    }
+
+    /**
+     * Problem Listener for Standings Pane.
+     * 
+     * @author pc2@ecs.csus.edu
+     * @version $Id$
+     */
+
+    // $HeadURL$
+    public class ProblemListenerImplementation implements IProblemListener {
+
+        public void problemAdded(ProblemEvent event) {
+            refreshStandings();
+        }
+
+        public void problemChanged(ProblemEvent event) {
+            refreshStandings();
+        }
+
+        public void problemRemoved(ProblemEvent event) {
+            refreshStandings();
+        }
+
+        public void problemRefreshAll(ProblemEvent event) {
+            refreshStandings();
+        }
+
+    }
+
+    /**
+     * 
+     * @author pc2@ecs.csus.edu
+     *
+     */
+    public class RunListenerImplementation implements IRunListener {
+
+        public void runAdded(RunEvent event) {
+            // TODO Auto-generated method stub
+            // ignore
+        }
+
+        public void refreshRuns(RunEvent event) {
+            refreshStandings();
+        }
+
+        public void runChanged(RunEvent event) {
+            // TODO Auto-generated method stub
+            if (event.getAction().equals(Action.CHANGED)) {
+                refreshStandings();
+            }
+        }
+
+        public void runRemoved(RunEvent event) {
+            // TODO Auto-generated method stub
```

Comment:
Or here.

Reply:
ancient auto generated comment

Reply:
Not ancient,  per the diffs this is  new class.

Reply:
ancient eclipse default comment

Reply:
pushed fix